spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/repository/ReleaseRepositoryCustom.java

```diff
@@ -36,6 +36,16 @@
 	Release findLatestRelease(String releaseName);
 
 	/**
+	 * Find the lasted in time, release object, by name whose status is neither unknown nor failed.
+	 * This release can be used for upgrade/rollback operations.
+	 * @param releaseName the name of the release
+	 * @return the Release object
+	 * @throws {@link ReleaseNotFoundException} if no latest Release (with the deployed/deleted status) for the given
```

Comment:
Might be good to throw SkipperException to have a customized message instead of `release not found` message here.